feeds/pypi/pypi.go

```diff
@@ -73,28 +85,93 @@ func fetchPackages() ([]*Package, error) {
 	return rssResponse.Packages, nil
 }
 
+func fetchCriticalPackages(packageList []string) ([]*Package, error) {
+	responseChannel := make(chan *Response)
+	errChannel := make(chan error)
+
+	for _, pkgName := range packageList {
+		go func(pkgName string) {
+			resp, err := httpClient.Get(fmt.Sprintf(packageURLFormat, pkgName))
+			if err != nil {
+				errChannel <- err
+				return
+			}
+			defer resp.Body.Close()
+			rssResponse := &Response{}
+			err = xml.NewDecoder(resp.Body).Decode(rssResponse)
+			if err != nil {
+				errChannel <- err
+				return
+			}
+
+			responseChannel <- rssResponse
+		}(pkgName)
+	}
+
+	pkgs := []*Package{}
+	var lastErr error
+	for i := 0; i < len(packageList); i++ {
+		select {
+		case response := <-responseChannel:
+			pkgs = append(pkgs, response.Packages...)
+		case err := <-errChannel:
+			lastErr = err
```

Comment:
As `Latest()` returns no packages if there's an error given from this func, it'd be quicker just to `return nil, err` immediately here if an error is detected. Unless there's inherent value in the last error, as opposed to the first.